halpybot/packages/seals/userinfo.py

```diff
@@ -19,42 +19,29 @@ async def whois(engine: Engine, subject):
         subject (str): The Seal's name being searched
 
     Returns:
-        (str): The details about the Seal, or an error string if not found.
+        (Seal): The Seal object
 
     """
-    # Set default values
-    u_id, u_cases, u_name, u_regdate, u_distant_worlds, u_distant_worlds_2 = (
-        None,
-        None,
-        None,
-        None,
-        None,
-        None,
-    )
     connection = engine.raw_connection()
```

Comment:
this can likely be used as a context manager.

Reply:
engine.raw_connection() does not have an `__enter__` and `__exit__` attribute, so it cannot be used as a context manager.

Reply:
why use a raw engine then :thinking:

Reply:
Required due to SQLAlchemy not playing nice with returned values for SPs. 

https://docs.sqlalchemy.org/en/20/core/connections.html#calling-stored-procedures-and-user-defined-functions